pkg/network/driver/nmstate/spec_test.go

```diff
@@ -0,0 +1,558 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2023 Red Hat, Inc.
+ *
+ */
+
+package nmstate_test
+
+import (
+	"github.com/onsi/ginkgo/v2"
+	"github.com/onsi/gomega"
+
+	"kubevirt.io/kubevirt/pkg/network/driver/procsys"
+
+	"kubevirt.io/kubevirt/pkg/network/driver/nmstate"
+	"kubevirt.io/kubevirt/pkg/pointer"
+)
+
+const (
+	dummyName = "dummy-red"
+
+	ip4Addr0   = "10.10.10.10"
+	ip4Prefix0 = 24
+
+	ip6Addr0   = "2001::1"
+	ip6Prefix0 = 64
+)
+
+// The test strategy is to setup and read the state of the network configuration through the nmstate API.
+// Then, assert that the returned (nmstate) status is indeed as expected.
+//
+// Unlike the strategy taken with the nmstate status unit tests, to use the drivers adapters,
+// the nmstate spec unit tests do not act on the drivers directly.
+
+var _ = ginkgo.Describe("NMState Spec interfaces", func() {
```

Comment:
Does it make sense to test setting MTU, MAC address and interface state (up / down)?